src/amplitude-client.js

```diff
@@ -1384,7 +1384,14 @@ var _generateApiPropertiesTrackingConfig = function _generateApiPropertiesTracki
  */
 AmplitudeClient.prototype._limitEventsQueued = function _limitEventsQueued(queue) {
   if (queue.length > this.options.savedMaxCount) {
-    queue.splice(0, queue.length - this.options.savedMaxCount);
+    const deletedEvents = queue.splice(0, queue.length - this.options.savedMaxCount);
+    deletedEvents.forEach((e) => {
+      if (type(e.callback) === 'function') {
+        e.callback(0, 'No request sent', {
+          reason: 'Event dropped because options.savedMaxCount exceeded. User may be offline or have a content blocker',
```

Comment:
nit: this is relatively long compared to our other `reason` strings. I appreciate the helpful direction though. How would you feel if this was in the util logs instead?

Reply:
(not necessarily saying it should be in the util logs since I don't know how visible that might be) - and leaving this as-is might be a good idea as well.

Reply:
I believe that it's only in the util logs, users won't be able to have failed upload event handling via callback